plugins/anomali_threatstream/komand_anomali_threatstream/connection/connection.py

```diff
@@ -41,3 +42,25 @@ def test(self):
             raise ConnectionTestException(preset=ConnectionTestException.Preset.INVALID_JSON, data=response.text)
 
         return {"connection": "successful"}
+
+    def send(self):
+        try:
+            return self.session.send(self.request.prepare(), verify=self.request.verify)
+        except Exception as e:
+            raise PluginException(
+                cause=f"The following exception was raised: {self.hide_api_key(str(e))}",
+                assistance="Please verify your ThreatStream server status and try again. "
+                "If the issue persists please contact support.",
+            ) from ConnectionError  # Suppresses the exception context from the original error that exposes API key
+
+    @staticmethod
+    def hide_api_key(string):
+        """
+        ThreatStream queries expose the API key as a URL query parameter. This method hides the API key using a regex that
+        substitutes with the replacement the first instance in string of a substring that matches pattern.
+        The pattern regex matches the api_key URL query parameter. It captures whether or not another parameter
+        follows the api_key value in the group named" "end" with an &. The replacement regex retains the end group.
+        """
+        pattern = r"api_key=([a-zA-Z0-9]+)(?P<end>\&|$)"
+        replacement = r"api_key=********\g<end>"
+        return re.sub(pattern, replacement, string)
```

Comment:
I remember the last solution we discussed was `raise <exception> from None` - is this solution newer (and working)?

Reply:
In testing, `from ConnectionError` only adds an additional exception to the stack of errors that are ultimately printed. Since this is unnecessary, and it may not always be the case that the original error was a ConnectionError, I reverted this to `from None`.